Not manage keys creation

rhoml opened the pull request:
We are running multiple puppetmasters and we would like to distribute eyaml keys using our own puppet repository for the public key and some other method for the private key. With this PR we can ensure this modules manage file permission on keys and directories.

rhoml committed:
Not manage keys creation

ghoneycutt commented on manifests/init.pp:
```diff
@@ -118,7 +126,7 @@
     group   => $keys_dir_group,
     mode    => $public_key_mode,
-    require => Exec['eyaml_createkeys'],
+    require => $requires,
```
Remove the require's on the files and use a `before` on the exec

ghoneycutt commented on manifests/init.pp:
```diff
@@ -31,6 +31,7 @@
   },
   $createkeys_path     = '/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin',
   $manage_eyaml_config = true,
+  $manage_keys         = true,
```
This needs to be documented in the README.md

ghoneycutt replied:
We need to validate our inputs and when dealing with booleans, handle the case of stringified booleans. This can be done with

``` puppet
if is_string($manage_keys) == true {
  $manage_keys_real = str2bool($manage_keys)
} else {
  $manage_keys_real = $manage_keys
}
validate_bool($manage_keys_real)
```

rhoml replied:
of we can just enforce it to be a bool right? I rather prefer people using the right values than giving them options to use strings

ghoneycutt replied:
We do enforce it with the code above. This is done because of Puppet's history of handling boolean values poorly. It aligns with the principal of least surprise, so if someone does pass a stringified boolean, things still work as expected.

ghoneycutt commented on manifests/init.pp:
```diff
@@ -105,11 +106,18 @@
-    command => "eyaml createkeys --pkcs7-private-key=${private_key_path} --pkcs7-public-key=${public_key_path}",
-    creates => $private_key_path,
-    require => File['eyaml_keys_dir'],
+  if $manage_keys {
```
`if manage_keys == true {`

rhoml replied:
This is not required since `if $manage_keys {`does this validation if we ensure it is a bool.

ghoneycutt commented on manifests/init.pp:
```diff
@@ -105,11 +106,18 @@
+    }
+  }
+
+  case $manage_keys {
```
This conditional could be removed by using `before`

ghoneycutt commented:
Thanks @rhoml for your work, this is a great idea! Could you please make the noted changes and I will be happy to help with the spec tests, unless you would like to take a stab at those.

rhoml commented:
Hey @ghoneycutt I've did a couple of things
- Added variable description with expected rtype on the init.pp
- Added some tests 
- Added pre-commit.com file useful for development
- Updated gitignore so we can use bundler
- Added the new minor version upgrade to the metadata.json file.

rhoml commented:
@ghoneycutt any news on this?

ghoneycutt commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,8 @@
+-   repo: git://github.com/pre-commit/mirrors-puppet-lint
```
what is this file?

rhoml replied:
This is a file to use pre-commit.com at development time, pretty useful for githooks management

ghoneycutt commented on metadata.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ghoneycutt-eyaml",
-  "version": "0.3.0",
+  "version": "0.3.1",
```
The maintainer will update this upon release. For instance, I might integrate other work into the release or merge ofter PR's first and release them, so you would not want to make these types of changes for version numbers.

ghoneycutt commented on manifests/init.pp:
```diff
@@ -21,7 +128,7 @@
   $config_dir_mode     = '0755',
   $config_ensure       = 'file',
-  $config_path         = '/etc/eyaml/config.yaml',
+  $config_path         = "${::settings::confdir}/config.yaml",
```
`settings` is a class.. this line should stay as it is, as you can already override with Hiera or a parameterized class

ghoneycutt commented on spec/classes/init_spec.rb:
```diff
@@ -1,7 +1,78 @@
 
-  context 'with defaults for all parameters' do
-    it { should contain_class('eyaml') }
+describe 'eyaml', :type => :class do
```
Thanks for spec tests!!

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,6 +2,113 @@
+# Here you should define a list of variables that this module would require.
+#
+# * `package_name`: hiera-eyaml requires the hiera-eyaml gem to be installed.
+#    rtype: string
```
why `rtype` instead of `type` or `types` ?

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,6 +2,113 @@
+#    rtype: integer
+#    Default: 0500
+#
+# * `public_key_path`: Absolute path to the public key.
```
thanks for documenting all these

ghoneycutt commented on Rakefile:
```diff
@@ -1,18 +1,10 @@
+PuppetLint.configuration.fail_on_warnings = true
+PuppetLint.configuration.send('relative')
 PuppetLint.configuration.send('disable_80chars')
-PuppetLint.configuration.relative = true
```
this needs to be here

ghoneycutt commented on Rakefile:
```diff
@@ -1,18 +1,10 @@
+PuppetLint.configuration.send('relative')
 PuppetLint.configuration.send('disable_80chars')
-PuppetLint.configuration.relative = true
+PuppetLint.configuration.send('disable_class_inherits_from_params_class')
```
these disables should be removed

ghoneycutt commented on Rakefile:
```diff
@@ -1,18 +1,10 @@
+PuppetLint.configuration.send('disable_single_quote_string_with_variables')
 PuppetLint.configuration.ignore_paths = ["spec/**/*.pp", "pkg/**/*.pp"]
-
-desc 'Validate manifests, templates, and ruby files'
```
why was this removed?

ghoneycutt commented on .sync.yml:
```diff
@@ -0,0 +1,9 @@
+---
```
what is this file for?

rhoml replied:
https://github.com/puppetlabs/modulesync_configs#defining-module-files

ghoneycutt replied:
You use ModuleSync? This is for Puppet Labs' modules to sync updates to all of their modules.

ghoneycutt commented on Gemfile:
```diff
@@ -1,15 +1,15 @@
 source ENV['GEM_SOURCE'] || 'https://rubygems.org'
 
-if puppetversion = ENV['PUPPET_GEM_VERSION']
```
This needs to be here so we can do matrix testing of different puppet versions

rhoml replied:
I'll fix this

ghoneycutt commented on Gemfile:
```diff
@@ -1,15 +1,15 @@
-  gem 'puppet', :require => false
-end
 
+gem 'rake'
```
these don't need to be here as they are just dependencies

ghoneycutt commented on Gemfile:
```diff
@@ -1,15 +1,15 @@
-gem 'puppetlabs_spec_helper', '>= 0.1.0'
-gem 'facter', '>= 1.7.0'
-gem 'rspec-puppet'
+gem 'rspec-puppet', '2.2.0'
```
why pin at that exact version? we want the newest

ghoneycutt commented on Gemfile:
```diff
@@ -1,15 +1,15 @@
-gem 'facter', '>= 1.7.0'
-gem 'rspec-puppet'
+gem 'rspec-puppet', '2.2.0'
+gem 'puppet', '3.8.4'
```
See comment above regarding matrix testing

rhoml replied:
I needed to test on our puppet version and forgot to roll it back

ghoneycutt commented on Gemfile:
```diff
@@ -1,15 +1,15 @@
-gem 'rspec-puppet'
+gem 'rspec-puppet', '2.2.0'
+gem 'puppet', '3.8.4'
+gem 'puppetlabs_spec_helper', '0.10.3'
```
again, we want the newest here

ghoneycutt commented on manifests/init.pp:
```diff
@@ -21,7 +128,7 @@
   $config_dir_group    = 'root',
   $config_dir_mode     = '0755',
   $config_ensure       = 'file',
-  $config_path         = '/etc/eyaml/config.yaml',
```
Why change the location of the eyaml config? I believe it should be in /etc/eyaml and not /etc/puppet or whatever confdir is.

rhoml replied:
I didn't meant to change the location just wanted the config file to be related to the config dir but i missed the it was using the class settings. Can we make it a local variable instead?

rhoml replied:
Config_dir is pointing to $::settings::confdir so it didn't make sense that the config_path was different right?

rhoml replied:
I made a change to make `$config_path = "${config_dir}/config.yaml"` I think it makes better sense now right? and the same applies for the public and private keys which should depend on the `keys_dir` variable

ghoneycutt replied:
config_path should stay the same because of the params accessing previous params as variables that I mentioned.

rhoml commented:
Removes sync.yml

rhoml commented:
Hey @ghoneycutt any new on this? think it can be merged now?

rhoml commented:
ping @ghoneycutt any news?

rhoml commented:
Hey @ghoneycutt can we get this PR merged?

rhoml commented:
@ghoneycutt is there any interest on merging this PR? otherwise I will have to fork the module

ghoneycutt commented on .rspec:
```diff
@@ -0,0 +1,4 @@
+--color
```
This is not needed.

ghoneycutt commented on Gemfile:
```diff
@@ -6,10 +6,16 @@ else
 gem 'puppetlabs_spec_helper', '>= 0.1.0'
 gem 'facter', '>= 1.7.0'
 gem 'rspec-puppet'
+gem 'rake', '~> 10.5.0'
```
Rake should be down in the ruby 1.8.7 conditional such as https://github.com/ghoneycutt/puppet-module-ssh/blob/master/Gemfile

ghoneycutt commented on Gemfile:
```diff
@@ -6,10 +6,16 @@ else
   gem 'puppet', :require => false
 end
 
+gem 'mocha'
```
These four are not needed

ghoneycutt commented on Rakefile:
```diff
@@ -1,5 +1,8 @@
 require 'puppetlabs_spec_helper/rake_tasks'
 require 'puppet-lint/tasks/puppet-lint'
+
+PuppetLint.configuration.fail_on_warnings = true
```
not sure these are actually needed

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,17 +2,124 @@
 #
 # Module to manage eyaml
 #
+# Variables
```
Appreciated, but I document all of the parameters in the README. Check out https://github.com/ghoneycutt/puppet-module-ssh

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,17 +2,124 @@
   $keys_dir_mode       = '0500',
-  $public_key_path     = '/etc/puppet/keys/public_key.pkcs7.pem',
-  $private_key_path    = '/etc/puppet/keys/private_key.pkcs7.pem',
+  $public_key_path     = "${keys_dir}/public_key.pkcs7.pem",
```
keep the hard coding of /etc/puppet/keys and Puppet now has issue with params using a previous param as a variable.

ghoneycutt commented on manifests/init.pp:
```diff
@@ -2,17 +2,124 @@
   $package_provider    = 'gem',
   $package_ensure      = 'present',
-  $keys_dir            = "${::settings::confdir}/keys",
+  $keys_dir            = '/etc/puppet/keys',
```
good catch

ghoneycutt commented:
I'm getting this ready for v1.0.0 and will be incorporating your changes. Thank you!

rhoml commented:
@ghoneycutt changes are updated now and rebased

rhoml commented:
Closing this PR as it is too far behind and I think you already implemented all the things proposed here anyways
